sqlite_query.c

```diff
@@ -104,24 +302,111 @@ sqlite_convert_to_pg(Oid pgtyp, int pgtypmod, sqlite3_stmt * stmt, int stmt_coli
 				 * "regular" process because its already implemented and
 				 * working properly.
 				 */
-				if (sqlite_value_affinity == SQLITE_INTEGER || sqlite_value_affinity == SQLITE_FLOAT)
-				{
-					double		value = sqlite3_column_double(stmt, stmt_colid);
-					Datum		d = DirectFunctionCall1(float8_timestamptz, Float8GetDatum((float8) value));
-
-					return (struct NullableDatum) { d, false};
-				}
-				else
+				switch (sqlite_value_affinity)
 				{
-					valstr = sqlite_text_value_to_pg_db_encoding(stmt, stmt_colid);
+					case SQLITE_INTEGER:
```

Comment:
We can read timestamp from bigint value directly like below, call `float8_timestamptz` for bigint value may not correct. Float8 cannot cover all value of int64 correctly
```c
Timestamp value = (Timestamp) sqlite3_column_int64(stmt, stmt_colid);
return (struct NullableDatum) {TimestampGetDatum(value), false};
```

Reply:
This is incorrect current implementation, see https://github.com/pgspider/sqlite_fdw/blob/90dad0a3ae7b8ffd129149088e094276c439ed61/sqlite_query.c#L107 Recommended patch have added, thanks for good idea, @nxhai98 !

Reply:
Thank for updating. Confirmed.

Reply:
Resolved.